src/TextArea.tsx

```diff
@@ -0,0 +1,164 @@
+import styled, {css} from 'styled-components'
+import React, {TextareaHTMLAttributes, ReactElement} from 'react'
+import {FormValidationStatus} from './utils/types/FormValidationStatus'
+import sx, {SxProp} from './sx'
+import {get} from './constants'
+
+export type TextAreaProps = {
+  /**
+   * Apply inactive visual appearance to the TextArea
+   */
+  disabled?: boolean
+  /**
+   * Indicates whether the TextArea must be checked
+   */
+  required?: boolean
+  /**
+   * Indicates whether the TextArea validation state
+   */
+  validationStatus?: FormValidationStatus
+  /**
+   * Block
+   */
+  block?: boolean
+  /**
+   * Allows resizing of the textarea
+   */
+  resize?: 'none' | 'both' | 'horizontal' | 'vertical'
+  /**
+   * Applies a custom width to the textarea
+   */
+  width?: number
+  /**
+   * Applies a custom height to the textarea
+   */
+  height?: number
+} & TextareaHTMLAttributes<HTMLTextAreaElement> &
+  SxProp
+
+const StyledTextArea = styled.textarea<TextAreaProps>`
```

Comment:
The Select and TextInput component both just use the private `TextInputWrapper` component to maintain consistent styling. I'd recommend we do that here too.

Reply:
I did think about this, and decided against it as that component had styling that wasn't relevant for this component, like warning state, size variants etc. Do we want to inherit those here? It feels like we'd be creating a dependency here for the sake of convenience, but potentially more detrimental to the user as we're loading unnecessary css and subsequently delivering a less performant component.

Reply:
You're right - the size variants aren't relevant. Most of the others are though:
| Styling prop | Used for TextArea? |
| ------------ | ------------------- |
| disabled | ✅ |
| hasLeadingVisual | ❌ |
| hasTrailingVisual | ❌ |
| block | ✅ |
| contrast | ✅ |
| validationStatus | ✅ |
| size | ❌ |

Reply:
I still think we should DRY this by using `TextInputWrapper` or something that replaces it. It's unlikely somebody is going to load the CSS for a `TextArea` and not also have a component on the page that loads `TextInputWrapper` styles (e.g.: `TextInput` or `Select`). Then, we'd downloading nearly identical styles twice.